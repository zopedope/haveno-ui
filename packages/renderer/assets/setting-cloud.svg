<svg fill="currentColor" width="58px" height="54px" viewBox="0 0 58 54" fill="none" xmlns="http://www.w3.org/2000/svg"> <path fill-rule="evenodd" clip-rule="evenodd" d="M43.8339 7.56984C41.5907 3.08354 36.9294 0 31.5427 0C25.2052 0 19.8717 4.2683 18.299 10.0676C17.5276 9.7233 16.672 9.53174 15.7714 9.53174C12.8205 9.53174 10.3534 11.588 9.74421 14.3362C9.47095 14.3111 9.19411 14.2983 8.91425 14.2983C3.99105 14.2983 0 18.2611 0 23.1493C0 28.0376 3.99105 32.0003 8.91425 32.0003C8.95701 32.0003 8.99969 32 9.0423 31.9994H45.1782L45.257 31.9996C52.0737 31.9996 57.5998 26.5128 57.5998 19.7444C57.5998 12.9761 52.0737 7.48926 45.257 7.48926C44.7757 7.48926 44.3008 7.51661 43.8339 7.56984Z" fill="currentColor"/> <path d="M26.2105 43.7896L22.1053 40.0001L18 43.7896" stroke="currentColor" stroke-width="2" stroke-linecap="round"/> <path d="M22.1052 52.316V40.2739" stroke="currentColor" stroke-width="2" stroke-linecap="round"/> <path d="M39.4713 48.8423L35.366 52.6318L31.2607 48.8423" stroke="currentColor" stroke-width="2" stroke-linecap="round"/> <path d="M35.366 40.3158V52.3579" stroke="currentColor" stroke-width="2" stroke-linecap="round"/></svg>