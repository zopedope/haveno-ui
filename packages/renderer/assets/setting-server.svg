<svg version="1.2" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 62" width="32" height="62">
	<title>server</title>
	<path id="Layer"  fill-rule="evenodd" fill="currentColor" d="m31.6 25.8v34.5c0 1-0.9 1.7-2 1.7h-27.6c-1.2 0-2-0.7-2-1.7v-34.5c0-1 0.8-1.7 2-1.7h27.6c1.1 0 2 0.7 2 1.7zm-6.9 8.8c0.6 0 1-0.3 1-0.8v-4.1c0-0.5-0.4-0.8-1-0.8h-17.8c-0.6 0-1 0.3-1 0.8v4.1c0 0.5 0.4 0.8 1 0.8zm1 4c0-0.5-0.4-0.8-1-0.8h-17.8c-0.5 0-0.9 0.3-0.9 0.8v4c0 0.5 0.4 0.9 0.9 0.9h17.8c0.6 0 1-0.4 1-0.9zm-11.8 10.2c0 1.1 0.8 2 1.9 2 1.1 0 2-0.9 2-2 0-1.1-0.9-1.9-2-1.9-1.1 0-1.9 0.8-1.9 1.9zm0 6.9c0 1.1 0.8 2 1.9 2 1.1 0 2-0.9 2-2 0-1.1-0.9-2-2-2-1.1 0-1.9 0.9-1.9 2z"/>
	<path id="Layer" stroke="currentColor" stroke-linecap="round" stroke-width="2" d="m5.4 5.8l4.1-3.8 4.1 3.8m-4.1 8.5v-12m17.3 8.5l-4.1 3.8-4.1-3.8m4.1-8.5v12.1"/>
</svg>