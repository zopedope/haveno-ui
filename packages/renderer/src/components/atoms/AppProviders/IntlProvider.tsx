// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
