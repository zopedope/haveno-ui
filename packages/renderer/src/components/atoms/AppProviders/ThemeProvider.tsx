// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
