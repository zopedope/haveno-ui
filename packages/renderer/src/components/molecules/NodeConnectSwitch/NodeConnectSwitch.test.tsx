// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import { NodeConnectSwitch } from ".";
import { AppProviders } from "@atoms/AppProviders";
import { ReactComponent as CloudIcon } from "@assets/setting-cloud.svg";
import { ReactComponent as ServerIcon } from "@assets/setting-server.svg";

describe("molecules::NodeConnectSwitch", () => {
  it("renders without exploding", () => {
    const { asFragment, unmount } = render(
      <AppProviders>
        <NodeConnectSwitch>
          <NodeConnectSwitch.Method
            active={true}
            current={true}
            tabKey="local-node"
            label="Local Node"
            icon={<ServerIcon width="32px" height="62px" />}
          >
            Local Node Content
          </NodeConnectSwitch.Method>

          <NodeConnectSwitch.Method
            tabKey="remote-node"
            label="Remote Node"
            icon={<CloudIcon width="58px" height="54px" />}
            active={false}
            current={false}
          >
            Remote Node Content
          </NodeConnectSwitch.Method>
        </NodeConnectSwitch>
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("pre-renders active tab", () => {
    const { unmount } = render(
      <AppProviders>
        <NodeConnectSwitch active="local-node">
          <NodeConnectSwitch.Method
            current
            tabKey="local-node"
            label="Local Node"
            icon={<ServerIcon width="32px" height="62px" />}
          >
            Local Node Content
          </NodeConnectSwitch.Method>

          <NodeConnectSwitch.Method
            tabKey="remote-node"
            label="Remote Node"
            icon={<CloudIcon width="58px" height="54px" />}
          >
            Remote Node Content
          </NodeConnectSwitch.Method>
        </NodeConnectSwitch>
      </AppProviders>
    );
    expect(screen.queryByText("Local Node Content")).toBeInTheDocument();
    unmount();
  });

  it("clicking a tab reveals its contents", () => {
    const { unmount } = render(
      <AppProviders>
        <NodeConnectSwitch>
          <NodeConnectSwitch.Method
            current
            tabKey="local-node"
            label="Local Node"
            icon={<ServerIcon width="32px" height="62px" />}
          >
            Local Node Content
          </NodeConnectSwitch.Method>

          <NodeConnectSwitch.Method
            tabKey="remote-node"
            label="Remote Node"
            icon={<CloudIcon width="58px" height="54px" />}
          >
            Remote Node Content
          </NodeConnectSwitch.Method>
        </NodeConnectSwitch>
      </AppProviders>
    );
    const tabs = screen.queryAllByRole("tab");
    expect(tabs).toHaveLength(2);
    expect(screen.queryByText("Remote Node Content")).not.toBeInTheDocument();
    fireEvent.click(tabs[1]);
    expect(screen.queryByText("Remote Node Content")).toBeInTheDocument();
    unmount();
  });
});
