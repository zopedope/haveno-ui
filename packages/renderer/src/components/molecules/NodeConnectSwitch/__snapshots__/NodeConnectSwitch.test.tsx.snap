// Vitest Snapshot v1

exports[`molecules::NodeConnectSwitch > renders without exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-15po0m8"
  >
    <div
      class="mantine-1d0mff5"
    >
      <button
        class="mantine-m2616v"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <div
          class="mantine-199rwtt"
        >
          <div
            class="mantine-b2d3ff"
          >
            Current
          </div>
          <div
            class="mantine-prfd4k"
          >
            <svg
              height="62px"
              viewBox="0 0 32 62"
              width="32px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                server
              </title>
              <path
                d="m31.6 25.8v34.5c0 1-0.9 1.7-2 1.7h-27.6c-1.2 0-2-0.7-2-1.7v-34.5c0-1 0.8-1.7 2-1.7h27.6c1.1 0 2 0.7 2 1.7zm-6.9 8.8c0.6 0 1-0.3 1-0.8v-4.1c0-0.5-0.4-0.8-1-0.8h-17.8c-0.6 0-1 0.3-1 0.8v4.1c0 0.5 0.4 0.8 1 0.8zm1 4c0-0.5-0.4-0.8-1-0.8h-17.8c-0.5 0-0.9 0.3-0.9 0.8v4c0 0.5 0.4 0.9 0.9 0.9h17.8c0.6 0 1-0.4 1-0.9zm-11.8 10.2c0 1.1 0.8 2 1.9 2 1.1 0 2-0.9 2-2 0-1.1-0.9-1.9-2-1.9-1.1 0-1.9 0.8-1.9 1.9zm0 6.9c0 1.1 0.8 2 1.9 2 1.1 0 2-0.9 2-2 0-1.1-0.9-2-2-2-1.1 0-1.9 0.9-1.9 2z"
                fill="currentColor"
                fill-rule="evenodd"
                id="Layer"
              />
              <path
                d="m5.4 5.8l4.1-3.8 4.1 3.8m-4.1 8.5v-12m17.3 8.5l-4.1 3.8-4.1-3.8m4.1-8.5v12.1"
                id="Layer"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="mantine-1wlt8yk"
          >
            Local Node
          </div>
        </div>
      </button>
      <button
        class="mantine-m2616v"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <div
          class="mantine-199rwtt"
        >
          <div
            class="mantine-prfd4k"
          >
            <svg
              fill="none"
              height="54px"
              viewBox="0 0 58 54"
              width="58px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M43.8339 7.56984C41.5907 3.08354 36.9294 0 31.5427 0C25.2052 0 19.8717 4.2683 18.299 10.0676C17.5276 9.7233 16.672 9.53174 15.7714 9.53174C12.8205 9.53174 10.3534 11.588 9.74421 14.3362C9.47095 14.3111 9.19411 14.2983 8.91425 14.2983C3.99105 14.2983 0 18.2611 0 23.1493C0 28.0376 3.99105 32.0003 8.91425 32.0003C8.95701 32.0003 8.99969 32 9.0423 31.9994H45.1782L45.257 31.9996C52.0737 31.9996 57.5998 26.5128 57.5998 19.7444C57.5998 12.9761 52.0737 7.48926 45.257 7.48926C44.7757 7.48926 44.3008 7.51661 43.8339 7.56984Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                d="M26.2105 43.7896L22.1053 40.0001L18 43.7896"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              />
              <path
                d="M22.1052 52.316V40.2739"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              />
              <path
                d="M39.4713 48.8423L35.366 52.6318L31.2607 48.8423"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              />
              <path
                d="M35.366 40.3158V52.3579"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="mantine-1wlt8yk"
          >
            Remote Node
          </div>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
